### PR TITLE
Config filter sample update

### PR DESCRIPTION
**Pull request by samsp-msft**
Renamed config sample to make it clearer what it does.
Added logic to look at destination addresses, and if they match a pattern, treat those as redirects to environment variables.

### CODE REVIEW

**Comment by samsp-msft** [comment]
> This makes for a good config filter demo, but I don't want to loose our general config demo.

Hmm, I must confess that I didn't realize that it was supposed to be a config demo 😺 - this is partly due to the lack of documentation, but also the config file in this sample did not seem to be particularly special.

We probably need to rethink what a config sample would look like, and how that would relate to the pretty exhaustive config that is now part of the config docs. I think whatever the sample includes, it needs to be much more fully documented! We should probably start with a list of requirements/goals for a config sample and then see which sample those fit in, or does it warrant one of its own.

**Comment by Tratcher** [comment]
Putting the config that you commented in https://github.com/microsoft/reverse-proxy/blob/main/docs/docfx/articles/configfiles.md#all-config-properties into a config sample seems like a good approach. I'd want the config filter sample to have a much more targeted config file.

**Comment by Tratcher** [comment]
Missing a using?
`samples/ReverseProxy.ConfigFilter.Sample/CustomConfigFilter.cs(27,64): error CS0103: (NETCORE_ENGINEERING_TELEMETRY=Build) The name 'StringComparer' does not exist in the current context`

File: samples/ReverseProxy.ConfigFilter.Sample/appsettings.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        "SessionAffinity": {
+          "Enabled": "true"
+        },
+        "HealthCheck": {
+          "Passive": {
+            "Enabled": "true",
+            "Policy": "TransportFailureRate",
+            "ReactivationPeriod": "00:05:00"
+          }
+        },
```
Comment (Tratcher) [inline samples/ReverseProxy.ConfigFilter.Sample/appsettings.json:62]: Trim out any optional config not related to the config filter, it's distracting.
```suggestion
```

File: samples/ReverseProxy.ConfigFilter.Sample/appsettings.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          {
+            "RequestHeader": "foo0",
+            "Append": "bar"
+          },
+          {
+            "ResponseHeader": "foo",
+            "Append": "bar",
+            "When": "Always"
+          }
+        ]
```
Comment (Tratcher) [inline samples/ReverseProxy.ConfigFilter.Sample/appsettings.json:47]:
```suggestion
```

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Looks like this is ready to merge.

**Comment by samsp-msft** [comment]
> Missing a using?
> `samples/ReverseProxy.ConfigFilter.Sample/CustomConfigFilter.cs(27,64): error CS0103: (NETCORE_ENGINEERING_TELEMETRY=Build) The name 'StringComparer' does not exist in the current context`

Yep, you suggested the change that added the comparer, that I accepted without merging locally 😺

**Comment by Tratcher** [comment]
#762

**Pull request merged**